agent/lib/internal_address_source.js

```diff
@@ -79,6 +79,22 @@ function same_allocation(a, b) {
     return true;
 }
 
+function same_messages(a, b) {
+    for (var i in a) {
+        var equal = false;
+        for (var j in b) {
+            if (a[i] === b[j]) {
+                equal = true;
+                break;
+            }
+        }
```

Comment:
Should ensure the 2 lists are the same length.

Reply:
Yes, you are right.  Your comment made me realise that there was a further defect in `same_allocation`. ` same_allocation(a, a')` would have returned erroneously true where `a` is  allocated to 1, 2 and `a' `is allocated to 1,2,3. My change makes the implementation symmetric.